### ISSUE TITLE
[Bug]: [Context Menu] When right-clicking the mouse, the default menu is triggered to be displayed 

### ISSUE THREAD

**Issue by pinky-pig** [radix-vue/shadcn-vue#260]
### Environment

```bash
Developement/Production OS: Windows 10 19043.1110
Node version: 16.0.0
Package manager: pnpm@8.6.0
Radix Vue version: 1.0.0
Shadcn Vue version: 1.0.0
Vue version: 3.0.0
Nuxt version: 3.0.0
Nuxt mode: universal
Nuxt target: server
CSS framework: tailwindcss@3.3.3
Client OS: Windows 10 19043.1110
Browser: Chrome 90.0.4430.212
```

### Link to minimal reproduction

https://www.shadcn-vue.com/docs/components/context-menu

### Steps to reproduce

![shadcn-vue-bug](https://github.com/radix-vue/shadcn-vue/assets/42307398/052e8156-287e-491b-a41a-e742d0cd840d)

When right-clicking the mouse, the default menu is triggered to be displayed. When the right-click menu of shadcn-vue context-menu is already displayed and you right-click the mouse again, if the mouse right-click is released quickly, the browser's default menu will be triggered.

### Describe the bug

After investigation, it was found that the default animation time of data-[state=closed] in shadcn-vue context-menu is 0.15s. If the context-menu is already open, when right-clicking again, the old context-menu will first close, requiring a 0.15s animation time, and then the new context-menu will be opened. However, if the mouse right-click is released within 0.15s, the browser's default context-menu will be triggered.

### Expected behavior

Perhaps it may not be a bug, but it still took me half an hour to find the reason. 
Possible optimizations include: 1. Reduce the animation time for "close," for example, I set it to 0.05, which works fine. 2. When opening, obtain information using document.getAnimations.
If I am causing any disturbance, please forgive my presumption.

**Comment by hrynevychroman** [comment]
I don't think this is a bug, for example if you right click on YouTube video, there is custom context menu shown, but when you clicked one more time, it will show you default one. I think this is because of accessibility, @zernonia sorry for tag, but need some final thought from you 🙂 

Attached some screenshots, don't look at text, it's my native language (ukrainian) 😁 
<img width="459" alt="image" src="https://github.com/radix-vue/shadcn-vue/assets/82209198/5c14e276-bdc9-4e25-93a1-b40377b7bc9d">
<img width="451" alt="image" src="https://github.com/radix-vue/shadcn-vue/assets/82209198/3c96335c-30e3-462f-bc80-69c37f0fc6b1">

For example on native context menu you can turn on (Picture in Picture) Arc feature, but YouTube don't give that option)

**Comment by pinky-pig** [comment]
It's not a bug to have the default menu appear again after the custom menu has already appeared😆

However, the issue here is that, after the appearance of the custom menu, right-clicking elsewhere may not always yield the same result. It's possible to get the default menu or the custom menu at a new location. The reason for this behavior is as described in the bug description above and depends on the time gap between right-clicking down and releasing.

Even with a repeated right-click, the outcome may vary, which can be confusing😂

After a custom menu has appeared, if you right-click and hold without releasing until the custom menu closes, and then release the right mouse button after the closing animation is complete, the new custom menu will appear in a new location. This is expected behavior. However, if the right mouse button is released while the closing animation of the previous custom menu is still ongoing, the new custom menu will not appear in a new location; instead, the default menu will appear.

In essence, the problem lies in the relationship between the opening of a new menu and the closing animation of the previous one.

 the release time > closing duration : 
 
![context-menu-normal](https://github.com/radix-vue/shadcn-vue/assets/42307398/253dfbd2-fea3-41fe-b630-6ebe51790e27)

 the release time < closing duration : 
 
![context-menu-bug](https://github.com/radix-vue/shadcn-vue/assets/42307398/76cc4dc9-34d1-4e2b-bc91-0858f8c2e9ff)

Youtube contextmenu :
 
![context-menu-youtube](https://github.com/radix-vue/shadcn-vue/assets/42307398/2e4c1c42-6148-4d41-9a84-cf40b9f00e39)

😆😘

**Comment by hrynevychroman** [comment]
Thanks for more details, I will try to watch and make a PR if find something ☺️

**Comment by hrynevychroman** [comment]
Need someone with Windows system, everything works fine on MacOS, can't reproduce it at all 😢

**Comment by pinky-pig** [comment]
I also cannot reproduce it on macOS indeed  🤣
And I have a Windows computer, and I'm willing to do whatever you need 😆

**Comment by hrynevychroman** [comment]
@pinky-pig Hello 👋 

Can you try to reproduce this bug on main [radix-vue website](https://www.radix-vue.com/components/context-menu.html#context-menu)? Look through code and think this issue is more to Radix-vue codebase, because here used all components from there, just added some styling)

btw, on main website there is no (default contextmenu) always opened custom one 🤔

**Comment by sadeghbarati** [comment]
Yes If I hit right click **fast** on that spot

**Comment by pinky-pig** [comment]
![f5b261db2f01d3655bf348a1adaf97f7](https://github.com/radix-vue/shadcn-vue/assets/42307398/329e3f0e-f4e2-4790-82a6-3609c1e0ad55)

I couldn't reproduce this bug on the radix-vue website, so I raised the issue here 🤣
Look through code, I initially thought the bug was caused by the 'data-[state=closed]' animate-out animation in shadcn-vue. But I haven't verified it yet 🥲

**Comment by pinky-pig** [comment]
![7d707e7f77a661de0d065febd33e5e06](https://github.com/radix-vue/shadcn-vue/assets/42307398/72751783-3f37-40a3-9fab-7944797ac060)

It seems that when I use a relatively small duration for the 'data-[state=closed]' animation, the custom menu keeps appearing.

<img width="1190" alt="image" src="https://github.com/radix-vue/shadcn-vue/assets/42307398/51796e59-518d-46ce-beb9-2db467d2773d">

🤣🤣

**Comment by pinky-pig** [comment]
Now I have verified my hypothesis 😆

As mentioned above, the issue of having two possible outcomes when right-clicking on Windows is due to the 'data-[state=open]' and 'data-[state=closed]' animations.

This situation did not occur on the radix-vue website example but did occur on the shadcn-vue website example. The difference between the two is the style animations.

I demonstrated this in the GIF below. With just the addition of two animations, the example from radix-vue, which originally did not have this bug on StackBlitz, exhibited an anomaly. (For the sake of demonstration, the duration here is set to 0.3s, whereas in shadcn-vue, it is 0.15s, which also causes this issue)

[The code in the GIF](https://stackblitz.com/edit/yqegdx?file=src%2FApp.vue)

![c8878fed7cc770d42d5bb1594f5f4fc0](https://github.com/radix-vue/shadcn-vue/assets/42307398/71549def-692b-49e4-8f01-b73a003ceb91)

In conclusion, because shadcn-vue only added some styles to radix-vue, the simplest way to address this issue is to minimize the duration of data-[state=closed] or remove the animation effect. Certainly, how to resolve this issue may need to be decided by shadcn-vue core members. 

Thanks for the open-source contribution😘. If there is a need for me to submit a pull request, I am also willing to do so 😉

**Comment by sadeghbarati** [comment]
[`shadcn-svelte` Context Menu](https://www.shadcn-svelte.com/docs/components/context-menu)


https://github.com/radix-vue/shadcn-vue/assets/17789047/c60c3e92-7e51-4025-8a08-765d28fb4502

**Comment by KelvinTee58** [comment]
你好，兄弟，我也出现这个问题了，你是把动画删掉就能解决问题了吗？

Hello brother, I also have this problem. Can you solve the problem by deleting the animation css?

and I also is macos

https://github.com/radix-vue/shadcn-vue/assets/32637205/3ab896b1-60ba-4389-a66b-6258cf0f4071

**Comment by KelvinTee58** [comment]
I got it. I just deleted the css " data-[state=closed]:animate-out " attribute in the "ContextMenuContent.vue" file and it works. I hope it works for you.

**Comment by brokuka** [comment]
any updates ?

**Comment by Ambitie** [comment]
It's been a year. Still no one cares this bug. Suggest to change the name of the project. This project has nothing to do with shadcn at all.

**Comment by zernonia** [comment]
Fixed with recent update to `reka-ui` 😁
